src/Controls/src/Core/VisualElementProfiler/VisualElementProfiler.cs

```diff
@@ -0,0 +1,255 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace Microsoft.Maui.Controls;
+
+internal class VisualElementProfiler
+{
+	class LayoutPassTypeStats : ILayoutPassTypeStats
+	{
+		public LayoutPassTypeStats(LayoutPassType layoutPassType, long count = 0, long totalTime = 0)
+		{
+			LayoutPassType = layoutPassType;
+			Count = count;
+			TotalTime = totalTime;
+		}
+
+		public LayoutPassType LayoutPassType { get; }
+		public long Count { get; set; }
+		public long TotalTime { get; set; }
+	}
+
+	class LayoutPassStats : ILayoutPassStats
+	{
+		private readonly LayoutPassTypeStats[] _layoutPassTypeStats =
```

Comment:
I think this can be static and shared across other instances

Reply:
This is not static on purpose, because they can eventually have more than one profiler active on different parts of the app.